Appparent use-after-free

The compiler says:
```
In file included from /usr/include/stdio.h:964,
                 from /builddir/build/BUILD/c-blosc2-2.9.1/blosc/frame.c:11:
In function 'fprintf',
    inlined from 'frame_get_vlmetalayers' at /builddir/build/BUILD/c-blosc2-2.9.1/blosc/frame.c:1645:9:
/usr/include/bits/stdio2.h:79:10: warning: pointer 'eframe_name_76' may be used after 'free' [-Wuse-after-free]
   79 |   return __fprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1, __fmt,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   80 |                         __va_arg_pack ());
      |                         ~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/c-blosc2-2.9.1/blosc/frame.c: In function 'frame_get_vlmetalayers':
/builddir/build/BUILD/c-blosc2-2.9.1/blosc/frame.c:1643:7: note: call to 'free' here
 1643 |       free(eframe_name);
      |       ^~~~~~~~~~~~~~~~~
```
and it seems to be right.

Also:
```
In function 'fprintf',
    inlined from 'sframe_open_chunk' at /builddir/build/BUILD/c-blosc2-2.9.1/blosc/sframe.c:59:7:
/usr/include/bits/stdio2.h:79:10: warning: pointer 'chunk_path_18' may be used after 'free' [-Wuse-after-free]
   79 |   return __fprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1, __fmt,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   80 |                         __va_arg_pack ());
      |                         ~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/c-blosc2-2.9.1/blosc/sframe.c: In function 'sframe_open_chunk':
/builddir/build/BUILD/c-blosc2-2.9.1/blosc/sframe.c:57:5: note: call to 'free' here
   57 |     free(chunk_path);
      |     ^~~~~~~~~~~~~~~~
```

--- a/blosc/sframe.c
+++ b/blosc/sframe.c
@@ -33,11 +33,9 @@ void* sframe_open_index(const char* urlpath, const char* mode, const blosc2_io *
       return NULL;
     }
     fp = io_cb->open(index_path, mode, io->params);
-    free(index_path);
-    if (fp == NULL) {
+    if (fp == NULL)
       BLOSC_TRACE_ERROR("Error creating index path in: %s", index_path);
-      return NULL;
-    }
+    free(index_path);
   }
   return fp;
 }
@@ -54,11 +52,9 @@ void* sframe_open_chunk(const char* urlpath, int64_t nchunk, const char* mode, c
       return NULL;
     }
     fp = io_cb->open(chunk_path, mode, io->params);
-    free(chunk_path);
-    if (fp == NULL) {
+    if (fp == NULL)
       BLOSC_TRACE_ERROR("Error opening chunk path in: %s", chunk_path);
-      return NULL;
-    }
+    free(chunk_path);
   }
   return fp;
 }

